viewing_party/party.py

```diff
@@ -1,23 +1,105 @@
-# ------------- WAVE 1 --------------------
-
 def create_movie(title, genre, rating):
-    pass
+    if title and genre and rating:
+        movie = {"title": title, "genre": genre, "rating": rating}
+    else: 
+        movie = None
+    return movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"] = [movie]
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"] = [movie]
+    return user_data
+
+def watch_movie(user_data, movie_title):
+    for movies in user_data["watchlist"]:
+        if movies["title"] == movie_title:
+            user_data["watched"].append(movies)
+            user_data["watchlist"].remove(movies)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    ratings_list = []
+    for movies in user_data["watched"]:
+                ratings_list.append(movies["rating"])
+    if ratings_list:
+        rating_average = sum(ratings_list)/len(ratings_list)
+    else:
+        rating_average = 0.0
+    return rating_average
 
-
+def get_most_watched_genre(user_data):
+    genres = []
+    for movies in user_data["watched"]:
+        genres.append(movies["genre"])
+    if genres:
+        most_watched_genre = max(genres, key = genres.count)

```

Comment:
This works, but there's a minor performance hit. `max` is an O(n) function and so is the `count` method. In this case `max` will end up calling `count` on every element in `genres`, bringing the whole complexity up to O(n^2). But there are ways to solve this in O(n) time, for example, using a frequency dictionary.

Now, as we hadn't even covered time complexity when you wrote this, this is not a major enough problem that I would move away from a Green. But I am pointing it out!